tensorflow/compiler/xla/mlir_hlo/lib/Dialect/mhlo/IR/hlo_ops.cc

```diff
@@ -5785,6 +5784,30 @@ LogicalResult ReduceOp::reifyReturnTypeShapes(
   return success();
 }
 
+LogicalResult ReduceOp::inferReturnTypeComponents(
+    MLIRContext* context, Optional<Location> location, ValueShapeRange operands,
+    DictionaryAttr attributes, RegionRange regions,
+    SmallVectorImpl<ShapedTypeComponents>& inferredReturnShapes) {
+
+  ReduceOp::Adaptor adaptor(operands, attributes, regions);
+  auto inputs = adaptor.inputs();
+  for (Value input : inputs) {
+    if(input == nullptr){
```

Comment:
Drop this conditional. input cannot be `nulllptr` here. All `inputs` of a reduce operation need to be defined ssa values.